code/modules/vending/_vending.dm

```diff
@@ -255,6 +259,27 @@
 	else
 		..()
 
+/obj/machinery/vending/update_appearance(updates=ALL)
+	. = ..()
+	if(machine_stat & BROKEN)
+		set_light(0)
+		return
+	set_light(powered() ? MINIMUM_USEFUL_LIGHT_RANGE : 0)
+
+
+/obj/machinery/vending/update_icon_state()
+	if(machine_stat & BROKEN)
+		icon_state = "[initial(icon_state)]-broken"
+		return ..()
+	icon_state = "[initial(icon_state)][powered() ? null : "-off"]"
+	return ..()
+
+
+/obj/machinery/vending/update_overlays()
+	. = ..()
+	if(light_mask && !(machine_stat & BROKEN) && powered())
+		. += emissive_appearance(icon, light_mask)
```

Comment:
update lum sources here (even though it has a light, it should be done anyway as a standard, I tried to generalise it when I coded it but emissive appearance needs to remain generic)

Reply:
addressed in 1389b29